src/main/kotlin/pl/newbies/lecture/application/LectureController.kt

```diff
@@ -81,6 +93,51 @@ fun Application.lectureRoutes() {
                     call.respond(lectureConverter.convert(updatedLecture))
                 }
 
+                put("/{id}/theme/image") {
+                    val id = call.parameters.getOrFail("id")
+                    val principal = call.principal<AresPrincipal>()!!
+                    val lecture = transaction { LectureDAO.findById(id)?.toLecture() }
+                        ?: throw LectureNotFoundException(id)
```

Comment:
this segment repeats  itself in every method

Reply:
I know, its frustrating but I don't think that extracting it would be a good idea. Even finding name for that 3-property object would be hard :/